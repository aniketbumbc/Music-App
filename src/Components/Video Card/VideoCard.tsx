import React from 'react';
import { useStyles } from '../../Style/Style';
import {
  Grid,
  Card,
  CardContent,
  CardMedia,
  Typography,
} from '@material-ui/core';
import { Video } from '../../Models/Model';

const VideoCard: React.FC<Video> = React.memo(({ video }) => {
  const { id, artist, title, release_year, image_url } = video;
  const classes = useStyles();
  return (
    <>
      <Grid item xs={12} sm={6} md={6} lg={3} key={id} data-testid='grid'>
        <Card className={classes.fullHeightCard}>
          <CardMedia
            data-testid='image'
            className={classes.musicMedia}
            image={image_url}
            style={{ width: '200px', height: '200px' }}
          />
          <CardContent
            className={classes.musicContent}
            data-testid='card-content'
          >
            <Typography gutterBottom variant='h5' component='h2'>
              {title}
            </Typography>
            <Typography variant='subtitle1' gutterBottom component='h5'>
              artitst: {artist}
            </Typography>
            <Typography variant='subtitle1' gutterBottom>
              Year: {release_year}
            </Typography>
          </CardContent>
        </Card>
      </Grid>
    </>
  );
});

export default VideoCard;
