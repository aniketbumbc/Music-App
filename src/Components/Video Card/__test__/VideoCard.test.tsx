import React from 'react';
import { render, screen } from '@testing-library/react';
import { video } from '../../../TestData/TestData';
import VideoCard from '../VideoCard';

describe('VideoCard', () => {
  it('Shoud have grid,image and card content in document', () => {
    render(<VideoCard video={video} />);

    const gridElement = screen.getByTestId('grid');
    const imageElement = screen.getByTestId('image');
    const content = screen.getByTestId('card-content');

    expect(gridElement).toBeInTheDocument();
    expect(imageElement).toBeInTheDocument();
    expect(content).toBeInTheDocument();
  });

  it('Shoud have heading element in document', () => {
    render(<VideoCard video={video} />);

    const h2Element = screen.getByRole('heading', { level: 2 });
    const h5Element = screen.getByRole('heading', { level: 5 });
    const h6Element = screen.getByRole('heading', { level: 6 });

    expect(h2Element).toBeInTheDocument();
    expect(h2Element).toHaveTextContent('Beautiful Trauma');
    expect(h5Element).toBeInTheDocument();
    expect(h6Element).toBeInTheDocument();
  });

  it('Shoud have correct object value element in document', () => {
    render(<VideoCard video={video} />);

    const h2Element = screen.getByRole('heading', { level: 2 });
    const h5Element = screen.getByRole('heading', { level: 5 });
    const h6Element = screen.getByRole('heading', { level: 6 });

    expect(h2Element).toHaveTextContent('Beautiful Trauma');
    expect(h5Element).toHaveTextContent('artitst: P!nk');
    expect(h6Element).toHaveTextContent('Year: 2017');
  });

  it('Shoud have incorrect object value element in document', () => {
    render(<VideoCard video={video} />);

    const h2Element = screen.getByRole('heading', { level: 2 });

    expect(h2Element).not.toHaveTextContent('artitst: P!nk');
  });

  it('Should render correct image', () => {
    render(<VideoCard video={video} />);

    const imageElement = screen.getByTestId('image');

    expect(imageElement).toHaveStyle({
      backgroundImage:
        'https://raw.githubusercontent.com/XiteTV/frontend-coding-exercise/679a82b1e7110c16e14412f1debaa118c10078a9/images/877825/images/app/w522_h292.jpg',
    });
  });

  it('Should take a snapshot of component', () => {
    const { asFragment } = render(<VideoCard video={video} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
