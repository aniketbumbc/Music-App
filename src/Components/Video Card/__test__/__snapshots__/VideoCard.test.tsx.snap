// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Should take a snapshot of component 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-3"
    data-testid="grid"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-fullHeightCard-49 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardMedia-root makeStyles-musicMedia-47"
        data-testid="image"
        style="background-image: url(https://raw.githubusercontent.com/XiteTV/frontend-coding-exercise/679a82b1e7110c16e14412f1debaa118c10078a9/images/877825/images/app/w522_h292.jpg); width: 200px; height: 200px;"
      />
      <div
        class="MuiCardContent-root makeStyles-musicContent-48"
        data-testid="card-content"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Beautiful Trauma
        </h2>
        <h5
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          artitst: P!nk
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          Year: 2017
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
